modules/integrating-with-github/proc-enabling-github-repository-discovery.adoc

```diff
@@ -0,0 +1,159 @@
+[id="enabling-github-repository-discovery"]
+= Enabling GitHub repository discovery
+
+By default, you register components manually into {product}.
+Consider configuring {product-short} to discover your GitHub repositories automatically.
+When a repository contains a `catalog-info.yaml` file, {product-short} ingests the entity into the catalog as a component.
+
+.Prerequisites
+* You link:{configuring-book-url}[added a custom {product-short} application configuration], and have sufficient permissions to modify it.
+* You have sufficient permissions in GitHub to create and manage a link:https://docs.github.com/en/apps/overview[GitHub App].
+
+.Procedure
+. To allow {product-short} to access the GitHub API, create a GitHub App.
+Opt for a GitHub App instead of an OAuth app to use fine-grained permissions, gain more control over which repositories the application can access, and use short-lived tokens.
+
+.. link:https://docs.github.com/en/apps/creating-github-apps/registering-a-github-app/registering-a-github-app[Register a GitHub App] with the following configuration:
++
+GitHub App name::
+Enter a unique name identifying your GitHub App, such as `integrating-with-rhdh-__<GUID>__`.
+
+Homepage URL::
+Enter your {product-short} URL: `pass:c,a,q[{my-product-url}]`.
+
+Authorization callback URL::
+Enter your {product-short} authentication backend URL: `pass:c,a,q[{my-product-url}/api/auth/github/handler/frame]`.
+
+Webhook URL::
+Enter your {product-short} URL: `pass:c,a,q[{my-product-url}]`.
+
+Webhook secret::
+Enter a strong secret.
```

Comment:
```suggestion
Webhook::
Uncheck "Active", as this is not needed for authentication and catalog providers. 
```
<img width="378" alt="image" src="https://github.com/user-attachments/assets/acd4c587-4c84-4582-9ee7-a945d2c5b4cb" />

Since upstream just marked this option as optional (see [changelogs](https://github.com/backstage/backstage/blob/master/packages/integration/CHANGELOG.md?plain=1#L117-L123) in BS1.38) and this is not used for auth or catalog provider, we can omit these 2 values. We just have to make sure that this is targeted for the RHDH v1.7 release.